test/amplitude-client.js

```diff
@@ -4149,4 +4199,253 @@ describe('AmplitudeClient', function () {
+      assert.equal(amplitude.options.serverZone, 'US');
+      amplitude.init(apiKey);
+      amplitude.setServerZone('invalid serverZone');
+      assert.equal(amplitude.options.serverZone, 'US');
+    });
+
+    it('should not set serverZone with invalid serverZoneBasedApi value', function () {
+      assert.equal(amplitude.options.serverZone, 'US');
+      amplitude.init(apiKey);
+      amplitude.setServerZone('EU', 'invalid serverZoneBasedApi');
+      assert.equal(amplitude.options.serverZone, 'US');
+    });
+
+    it('should set serverZone to EU', function () {
+      assert.equal(amplitude.options.serverZone, 'US');
+      amplitude.init(apiKey);
+      amplitude.setServerZone('EU');
+      assert.equal(amplitude.options.serverZone, 'EU');
+    });
+
+    it('should set serverZont to EU and serverUrl to EVENT_LOG_EU_URL with serverZoneBasedApi is true', function () {
+      assert.equal(amplitude.options.serverZone, 'US');
+      amplitude.init(apiKey);
+      amplitude.setServerZone('EU', true);
+      assert.equal(amplitude.options.serverZone, 'EU');
+      assert.equal(amplitude.options.apiEndpoint, 'api.eu.amplitude.com');
+    });
+
+    it('should set serverZont to EU and keep the default serverUrl with serverZoneBasedApi is false', function () {
+      assert.equal(amplitude.options.serverZone, 'US');
+      amplitude.init(apiKey);
+      amplitude.setServerZone('EU', false);
+      assert.equal(amplitude.options.serverZone, 'EU');
+      assert.equal(amplitude.options.apiEndpoint, 'api.amplitude.com');
+    });
+  });
+
+  describe('getUserId', function () {
+    beforeEach(function () {
+      reset();
+    });
+
+    it('should get userId', function () {
+      amplitude.init(apiKey, userId);
+      let currentUserId = amplitude.getUserId();
+      assert.equal(amplitude.options.userId, currentUserId);
+    });
+
+    it('should get userId null', function () {
+      amplitude.init(apiKey);
+      assert.equal(amplitude.getUserId(), null);
+    });
+  });
+
+  describe('getDeviceId', function () {
+    beforeEach(function () {
+      reset();
+    });
+
+    it('should get a random deviceId', function () {
+      amplitude.init(apiKey, userId);
+      assert.lengthOf(amplitude.getDeviceId(), 22);
+    });
+
+    it('should get deviceId', function () {
+      const currentDeviceId = 'aa_bb_cc';
+      amplitude.init(apiKey, null, { deviceId: currentDeviceId });
+      assert.equal(amplitude.getDeviceId(), currentDeviceId);
+    });
+  });
+
+  describe('setMinTimeBetweenSessionsMillis', function () {
+    beforeEach(function () {
+      reset();
+    });
+
+    it('should not set sessionTimeout with invalid input', function () {
+      amplitude.init(apiKey);
+      let newSessionTimeOut = 'invalid sessionTimeOut';
+      amplitude.setMinTimeBetweenSessionsMillis(newSessionTimeOut);
+      assert.equal(amplitude.options.sessionTimeout, 30 * 60 * 1000);
```

Comment:
nit: can get current value first, then assert two variables equal. Just hardcoded is not straightforward in test.